src/client/common/client_common.cpp

```diff
@@ -104,6 +104,15 @@ std::shared_ptr<grpc::Channel> create_channel_and_validate(const std::string& se
     mp::PingReply reply;
     auto status = stub.ping(&context, request, &reply);
 
+    // If the daemon cannot be connected to, throw here to keep the client from doing anything else and not
+    // create a new certificate.  The client will show an "unhandled exception", but that is OK in this
+    // instance since it's addressing a corner case and this code path will only run until the client
+    // establishes its cert for the daemon authentication. This whole function will be deprecated in the future.
+    if (status.error_code() == grpc::StatusCode::UNAVAILABLE)
```

Comment:
Could we instead reverse this logic, and only _not_ throw when connected, but unauthorized?

Reply:
Well, we want to return the channel pointer if the status is `OK` meaning it can connect to the daemon and is authorized and we want it to return `nullptr` if it can connect but `UNAUTHENTICATED`.  I guess what you are meaning is you want it to `throw` in all other cases, right?

Reply:
I just mean we should throw in any but the two cases we handle below. As is, it won't throw if `status.error_code()` is something else than `UNAVAILABLE`, but still neither OK or unauthenticated.

Reply:
Right, that's what I was *trying* to ask :wink:  I'll fix it up to reflect that.  Thanks!

Reply:
Ok, I believe I covered this better.